File: server/src/main/java/com/ps/server/dto/MoveResponseDTO.java
package com.ps.server.dto;

import com.ps.server.Logic.Change;
import lombok.*;

import javax.validation.constraints.NotNull;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class MoveResponseDTO {

    @NotNull
    public boolean wasMoveValid;

    public List<Change> listOfChanges;


    @Override
    public String toString() {
        return "MoveResponseDTO{" +
                "listOfChanges=" + listOfChanges +
                '}';
    }

    public boolean isWasMoveValid() {
        return wasMoveValid;
    }

    public void setWasMoveValid(boolean wasMoveValid) {
        this.wasMoveValid = wasMoveValid;
    }

    public List<Change> getListOfChanges() {
        return listOfChanges;
    }

    public void setListOfChanges(List<Change> listOfChanges) {
        this.listOfChanges = listOfChanges;
    }
}

File: client/src/app/DTO/moveResponseDTO.ts
import {ChangeDTO} from "./changeDTO";


/** @ignore*/
export class MoveResponseDTO {
    /** */
    public wasMoveValid: boolean;
    /** */
    public listOfChanges: ChangeDTO[];

    /** @ignore*/
    constructor(wasMoveValid: boolean, listOfChanges: ChangeDTO[]) {
        this.wasMoveValid = wasMoveValid;
        this.listOfChanges = listOfChanges;
    }
}